server/controller/UsersCtrl.js

```diff
@@ -0,0 +1,199 @@
+      } else {
+        res.status(401)
+            .send({ message: 'Authentication failed!' });
+      }
+    }).catch(() => {
+    });
+  },
+
+  /**
+   * logout - Logs out a user
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  logout(req, res) {
+    db.invalidToken.create({ token: req.headers['x-access-token']
+    || req.headers.authorization });
+    db.invalidToken.destroy({ where: {
+      createdAt: { $lt: new Date() - (48 * 60 * 60 * 1000) } } });
+    res.send({ message: `User with id:${req.decoded.UserId} logged out` });
+  },
+
+
+  /**
+   * allUsers - Gets all user details
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  allUsers(req, res) {
+    db.Users.findAll({ fields: [
+      'id',
+      'username',
+      'fullName',
+      'email',
+      'RoleId',
+      'createdAt',
+      'updatedAt'
+    ] })
+      .then((usersList) => {
+        res.send(usersList);
+      });
+  },
+
+
+  /**
+   * getUser - Get a single user based on email or username
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  getUser(req, res) {
+    db.Users.findOne({
+      where: {
+        $or: [{ email: req.params.id },
+          { username: req.params.id }]
+      }
+    }).then((user) => {
+      if (!user) {
+        return res.status(401)
+        .send({ message: `User ${req.params.id} cannot be found` });
+      }
+      res.send(user);
+    });
+  },
+
+  /**
+   * updateUser - Update user details
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  updateUser(req, res) {
+    db.Roles.findById(req.decoded.RoleId)
+      .then((role) => {
+        if (role.title === 'Admin'
+        || String(req.decoded.UserId) === req.params.id) {
+          db.Users.find({ where: {
+            id: req.params.id } })
+            .then((user) => {
+              user.update(req.body)
+                .then(updatedUser => res
+                  .send({ message: `${req.params.id} updated`,
+                    data: userDetails(updatedUser)
+                  }));
+            });
+        } else {
+          return (res.status(401)
+         .send({ message: 'Unauthorized Access' }));
+        }
+      });
+  },
+
+  /**
+   * deleteUser - Delete a user
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  deleteUser(req, res) {
+    db.Roles.findById(req.decoded.RoleId)
+      .then((role) => {
+        if (role.title === 'Admin') {
+          db.Users.find({ where: {
+            id: req.params.id } })
+      .then((user) => {
+        user.destroy()
+      .then(() => {
+        res.status(201).send({ message: `${req.params.id} has been deleted` });
```

Comment:
Wrong status code: 201 means created. 200 would be more appropriate. Also, add status codes for the rest of your responses in this controller that do not have one.